### PR TITLE
Zcurve

### LINKED ISSUE #21: Nutrition zscore plots
For nutrition surveys, we need plots that compare the observed z-scores to the WHO Standard. 

Expected output from an actual survey:
screen shot 2019-01-11 at 13 09 22

We should use ggplot2. Here is some quick sample code that uses the `anthro` package to generate the z-scores. But the function should really just take a vector of z-scores and necessary labels.

Quick illustration:
```R
library(ggplot2)
library(anthro)

n <- 1000
res <- anthro_zscores(sex = sample(c(1, 2), n, TRUE),
                  age = as.integer(rnorm(n, 1000, 3)),
                  lenhei = rnorm(n, 94, 3))


ggplot(res[res$flen == 0, ]) + 
  geom_density(aes(x = zlen), color = "darkred") + 
  stat_function(fun = dnorm, args = list(mean = 0, sd = 1), color = "darkgreen") +
  scale_x_continuous(limits = c(-6, 6))
```
screen shot 2019-01-11 at 14 25 59

Please use colorblind and printer friendly colors

### PR DESCRIPTION
**Pull request by zkamvar**
This will fix #21 

``` r
dat <- rnorm(204) + runif(1) # slightly skewed
zcurve(dat)
```

<sup>Created on 2019-01-15 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Currently, the only argument to this function is a vector of numeric data. It will check if it's numeric and then plot it.

### CODE REVIEW

File: R/zcurve.R [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#' @examples
+#' dat <- rnorm(204) + runif(1) # slightly skewed
+#' zcurve(dat)
+zcurve <- function(zscore) {
+
+  stopifnot(is.numeric(zscore))
+  dat <- data.frame(observed = zscore)
+  obs <- quote(observed)
+  ggplot(dat) +
+    geom_density(aes(x = !!obs, color = "observed")) +
```
Comment (dirkschumacher) [inline R/zcurve.R]: Isn't there `aes_string` for that?

File: R/zcurve.R [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#' Create a curve comparing observed Z-scores to the WHO standard.
+#'
+#' @param zscore a numeric vector containing computed zscores
+#' @return a ggplot2 object
```
Comment (dirkschumacher) [inline R/zcurve.R]: Maybe add a line that users can extend that object.

File: R/zcurve.R [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#' @export
+#' @importFrom ggplot2 ggplot aes stat_function geom_density scale_x_continuous labs
+#' @examples
+#' dat <- rnorm(204) + runif(1) # slightly skewed
+#' zcurve(dat)
```
Comment (dirkschumacher) [inline R/zcurve.R]: Can you add an example that sets a title for plot? E.g. "Weight-for-Height z-scores"?

**Comment by dirkschumacher** [comment]
Awesome thanks.

File: R/zcurve.R (hunk 1 above)
Comment (zkamvar) [reply R/zcurve.R]: Yes there is, but the function is [soft deprecated](https://ggplot2.tidyverse.org/reference/aes_.html#life-cycle), so I would rather go with the recommended version

**Comment by zkamvar** [comment]
``` r
library("epireports")
library("ggplot2")
dat <- rnorm(204) + runif(1) # slightly skewed
zcurve(dat) +
  labs(title = "Weight-for-Height Z-scores") +
  theme_classic()
```



<sup>Created on 2019-01-15 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

File: R/zcurve.R (hunk 2 above)
Comment (zkamvar) [reply R/zcurve.R]: Done

File: R/zcurve.R (hunk 3 above)
Comment (zkamvar) [reply R/zcurve.R]: Yup, see https://github.com/R4EPI/epireports/pull/25#issuecomment-454296280

**Pull request merged**